src/custom-elements/src/components/text-input/text-input.tsx

```diff
@@ -0,0 +1,149 @@
+import { Component, Element, Method, Prop, State, Watch, h } from '@stencil/core';
+import { CHI_STATES, ChiStates } from '../../constants/states';
+import { ICON_COLORS, IconColors } from '../../constants/color';
+import { TEXT_INPUT_SIZES, TextInputSizes } from '../../constants/size';
+import { TEXT_INPUT_TYPES, TextInputTypes } from '../../constants/constants';
+
+@Component({
+  tag: 'chi-text-input',
+  styleUrl: 'text-input.scss',
+  scoped: true
+})
+
+export class TextInput {
+  @Element() el: HTMLInputElement;
```

Comment:
@janibolkvadze, I'm just curious: is this the right type?

Reply:
I think we should change it to `HTMLElement` since the actual `HTMLInputElement` (`<input>`) is inside `<chi-text-input>`.
Also, I just noticed that it should have a prop `name` to define name attribute of the `<input>`.
What do you think?

Reply:
@janibolkvadze I think the right type is HtmlTextInputElement, that stencil creates that type dynamically. I'm talking by memory, so maybe it is slightly different.